test/reduce_test.cpp

```diff
@@ -589,7 +618,22 @@ struct verify_reduce_no_indices
         return (res);
     }
 
-    tensor<T> gpu() const
+    tensor<float> gpu() const
```

Comment:
[Q] Why can't we use `T`? Are you sure that it is Ok to use `float` tensor for verification of `double` reduce results?

Reply:
The reason is that in test/driver.cpp,  the verify() function use 
`double threshold = std::numeric_limits<value_type>::epsilon() * tolerance;`    to calculate the threshold,  in which the epsilon() depends on the data type,  this make our codes very difficult to use threshold since the epsilons for half, float and double are very different

Reply:
Ok!